internal/mage/engine.go

```diff
@@ -223,18 +234,16 @@ func (t Engine) test(ctx context.Context, race bool) error {
 		tests = tests.WithEnvVariable("_EXPERIMENTAL_DAGGER_CACHE_CONFIG", cacheEnv)
 	}
 
-	output, err := tests.
+	_, err = tests.
 		WithMountedFile(cliBinPath, util.DaggerBinary(c)).
 		WithEnvVariable("_EXPERIMENTAL_DAGGER_CLI_BIN", cliBinPath).
 		WithEnvVariable("_EXPERIMENTAL_DAGGER_RUNNER_HOST", endpoint).
 		WithMountedDirectory("/root/.docker", util.HostDockerDir(c)).
+		WithExec([]string{"go", "install", "gotest.tools/gotestsum@v1.10.0"}).
 		WithExec(args).
-		Stdout(ctx)
-	if err != nil {
-		return err
-	}
-	fmt.Println(output)
-	return nil
+		WithExec([]string{"gotestsum", "tool", "slowest", "--jsonfile=./tests.log", "--threshold=1s"}).
```

Comment:
This is super cool and helpful 🚀 The top two slowest tests in the most recent run on this PR, `TestPlatformCrossCompile` and `TestClientWaitsForEngine`, don't have `t.Parallel()` but could. IIRC there used to be problems with that in the past due to how the registry was setup, but that should no longer be an issue.

Feel free to update this PR with that, or I can send out a separate PR if you want to not bloat this change. I wouldn't be surprised if doing that makes a huge dent in the execution time of the tests since `TestPlatformCrossCompile` is probably just slow because it takes a long time to pull `crazymax/goxx:latest` and `TestClientWaitsForEngine` is slow just because it inherently has to wait a while, so it's not like either is slow due to lots of computation.

Reply:
Sounds good - done! And moved the `go install` earlier.